src/python/pants/engine/fs.py

```diff
@@ -66,12 +66,26 @@ def __repr__(self) -> str:
         )
 
 
+@dataclass(frozen=True)
+class Directory:
+    """The path to a directory."""
```

Comment:
I might note specifically that this is as opposed to `FileContent`, which will for example create parent directories as needed (is that correct?) so people don't think they need to use `Directory` except for when they *only* want to create a directory.

Reply:
Ok, I see that's described in `CreateDigest`. I personally would add the redundant text because this is higher up in the file but I don't believe it's necessary.